src/Heap.js

```diff
@@ -98,41 +96,45 @@ export default class Heap {
     }
 
     compare(a, b) {
-        if (a && b) {
-            if (Array.isArray(a) && Array.isArray(b)) {
-                if (a[0] > b[0] || (a[0] === b[0] && a[1] > b[1])) {
-                    return 1;
-                }
-
-                return -1;
+        const aKey = this.heap[a].key;
+        const bKey = this.heap[b].key;
+        if (Array.isArray(aKey) && Array.isArray(bKey)) {
+            if (aKey[0] > bKey[0] || (aKey[0] === bKey[0] && aKey[1] > bKey[1])) {
+                return 1;
```

Comment:
we did this a better way earlier can we do it again 🙁

Reply:
🐇